dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -755,6 +756,29 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks)
 
     auto disks = storage_policy->getDisks();
 
+    if (getStoragePolicy()->getName() != "default")
+    {
+        /// Check extra parts at different disks, in order to not allow to miss data parts at undefined disks.
+        std::unordered_set<String> defined_disk_names;
+        for (const auto & disk_ptr : disks)
+            defined_disk_names.insert(disk_ptr->getName());
+
+        for (auto & [disk_name, disk_ptr] : global_context.getDiskSelector().getDisksMap())
+        {
+            if (defined_disk_names.count(disk_name) == 0 && Poco::File(getFullPathOnDisk(disk_ptr)).exists())
+            {
+                for (Poco::DirectoryIterator it(getFullPathOnDisk(disk_ptr)); it != end; ++it)
+                {
+                    /// Skip temporary directories.
+                    if (startsWith(it.name(), "tmp"))
```

Comment:
Why we skip especially `tmp` directories? What if we have `/detached` directory or parts with other prefixes? https://github.com/clickhouse/ClickHouse/blob/master/dbms/src/Storages/MergeTree/MergeTreeDataPart.cpp#L394

Reply:
+